doc/reference/reference_lua/box_error/error_object.rst

```diff
@@ -6,59 +6,186 @@ error_object
 
 .. class:: error_object
 
-    Since version :doc:`2.4.1 </release/2.4.1>`.
-    Errors can be organized into lists. To achieve this, a Lua table representing an
-    error object has ``.prev`` field and ``e:set_prev(err)`` method.
+    An object that defines an error.
+    ``error_object`` is returned by the following methods:
+
+    *  :ref:`box.error.new() <box_error-new>`
+    *  :ref:`box.error.last() <box_error-last>`
+
+    .. _box_error-unpack:
+
+    .. method:: unpack()
+
+        Get error details that may include an error code, type, message, and trace.
+
+        **Example**
+
+        ..  literalinclude:: /code_snippets/test/errors/unpack_clear_error_test.lua
+            :language: lua
+            :start-after: Get error details: start
+            :end-before: Get error details: end
+            :dedent:
+
+        .. NOTE::
+
+            Depending on the error type, error details may include other attributes, such as :ref:`errno <box_error-errno>` or :ref:`reason <box_error-reason>`.
+
+    .. _box_error-raise:
+
+    .. method:: raise()
+
+        Raise the current error.
+
+        **See also:** :ref:`Raising an error <box_error_raise_error>`
+
+    .. _box_error-set_prev:
+
+    .. method:: set_prev(error_object)
+
+        **Since:** :doc:`2.4.1 </release/2.4.1>`
+
+        Set the previous error for the current one.
+
+        :param error_object body: an error object
+
+        **See also:** :ref:`Error lists <box_error_error_lists>`
 
     .. _box_error-prev:
 
     .. data:: prev
 
-        Return a previous error, if any.
+        **Since:** :doc:`2.4.1 </release/2.4.1>`
 
-    .. _box_error-set_prev:
+        Get the previous error for the current one.
+
+        :rtype: error_object
+
+        **See also:** :ref:`Error lists <box_error_error_lists>`
+
+    .. _box_error-code:
+
+    .. data:: code
+
+        The error code.
+        This attribute may return a :ref:`custom error <box_error_raise_custom_table_error>` code or a :ref:`Tarantool error <box_error_raise_tarantool_error>` code.
+
+        :rtype: number
+
+    .. _box_error-type:
+
+    .. data:: type
+
+        The error type.
+
+        :rtype: string
+
+        **See also:** :ref:`Custom error <box_error_raise_custom_error>`
+
+    .. _box_error-message:
+
+    .. data:: message
+
+        The error message.
+
+        :rtype: string
+
+    .. _box_error-trace:
+
+    .. data:: trace
+
+        The error trace.
+
+        :rtype: table
+
+    .. _box_error-errno:
+
+    .. data:: errno
+
+        If the error is a system error (for example, a socket or file IO failure),
+        returns a C standard error number.
+
+        :rtype: number
+
+    .. _box_error-reason:
+
+    .. data:: reason
+
+        **Since:** :doc:`2.10.0 </release/2.10.0>`
+
+        Returns the :ref:`box.info.ro_reason <box_introspection-box_info>` value at the moment of throwing the ``box.error.READONLY`` error.
```

Comment:
Maybe use a unified note or intro phrase for all parameters that apply only to readonly error?
"Only for box.error.READONLY:"